posthog/api/test/test_authentication.py

```diff
@@ -950,3 +955,83 @@ def test_after_timeout_modifications_require_reauthentication(self):
 
             res = self.client.get("/api/organizations/@current")
             assert res.status_code == 200
+
+
+class TestProjectSecretAPIKeyAuthentication(APIBaseTest):
+    def setUp(self):
+        super().setUp()  # Call the setup from APIBaseTest
+        self.team.secret_api_token = "phs_JVRb8fNi0XyIKGgUCyi29ZJUOXEr6NF2dKBy5Ws8XVeF11C"
+        self.team.save()
+        self.factory = APIRequestFactory()  # Use APIRequestFactory instead of RequestFactory
+
+    def test_authenticate_with_valid_secret_api_key_in_header(self):
+        # Simulate a request with a valid secret API key
+        wsgi_request = self.factory.get(
```

Comment:
You can also do `CONFIG_AUTO_LOGIN` which clears the session so you can continue to use the normal client. Feels like it would be simpler and this is what we do elsewhere for testing api keys for example [here](https://github.com/PostHog/posthog/blob/master/posthog/api/test/test_personal_api_keys.py#L231)

Reply:
Correct me if I'm wrong, but I'm not sure that applies to these tests because I'm testing the authenticator directly. I need a request to pass into the `authenticate` method.

I do have other tests that test the authentication in the context of the full request pipeline.